core/src/main/java/org/apache/calcite/sql/SqlSelect.java

```diff
@@ -111,7 +131,8 @@ public SqlSelect(SqlParserPos pos,
 
   @Override public List<SqlNode> getOperandList() {
     return ImmutableNullableList.of(keywordList, selectList, from, where,
-        groupBy, having, qualify, windowDecls, orderBy, offset, fetch, hints);
+        groupBy, having, qualify, windowDecls, orderBy, offset, fetch,
+        hints, topN);
```

Comment:
It seems to me in the design doc, the discussion of the order of the operand does not align with the change you have here. You would either need to update the design doc, or move topN to the correct order in the code.

Reply:
I will update the design doc. It seems the order is not crucial to unparsing, so I am going with Drew's suggestion of minimizing the chances of breaking anything by adding to the end.